stdlib/InteractiveUtils/test/runtests.jl

```diff
@@ -960,3 +960,13 @@ end # module
     using .OuterModule
     @test_nowarn subtypes(Integer);
 end
+
+@testset "Lowering behavior for keyword arguments" begin
+    f_with_kw(; T = Int) = T
+    _, rt = @code_typed f_with_kw()
+    @test rt === Type{Int}
+    _, rt = @code_typed f_with_kw(; T::Type{Int})
+    @test rt === Type{Int}
```

Comment:
I think it's better if we make the behavior accurately mimic what `kwfunc handling` does. Even if it seems like we can infer `::Type{Int}`, if the actual inference ends up being `DataType`, then `@code_typed` should probably follow that same behavior. This would help us avoid making wrong assumptions.

Reply:
If we can infer `Type{Int}`, we would return a `PartialStruct` preserving that, so it's pretty hard to say what it should mean exactly.